liferay-content-management/liferay-portal
# LPD-27595 Upgrade process

- Story: https://liferay.atlassian.net/browse/LPD-25799
- Task: https://liferay.atlassian.net/browse/LPD-27957

# What is this trying to solve?

Upgrade configuration from system to company for the customers that have autogenerate structure key and templated key configurations configured.


# How can you verify that it works?

Merge code with [this pr](https://github.com/liferay-content-management/liferay-portal/pull/5531) and deploy affected modules.

You can test it in two different ways:

1. Run the included tests.
2. Configure autogenerate keys properties  and stop bundle. Add schema.module.build.auto.upgrade to portal property and enable FF LPD-25799. Start portal again and see that old system properties are now configured how were they going to journal web/ instance settings.
